### PR TITLE
[exporter/azuredataexplorer] Status Attributes added to exported traces

### PR DESCRIPTION
**Pull request by asaharn**
**Description:** <Describe what has changed.>

Added an optional column in the exported trace data to store the status code and message as a dynamic field.

**Link to tracking Issue:** <Issue number if applicable>
[#26496](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/26496)

**Testing:** <Describe what testing was performed and which tests were added.>
Performed E2E ingestion tests and added Test Cases for new fields.

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: exporter/azuredataexplorerexporter/README.md [hunk 1]
```diff
@@ -92,7 +92,8 @@ This exporter maps OpenTelemetry  [trace](https://opentelemetry.io/docs/referenc
 | SpanId                    | A valid span identifier is an 8-byte array with at least one non-zero byte.                      |
 | ParentId                    | A parent spanId, for the current span                                                          |
 | SpanName                    | The span name                                                                                  |
-| SpanStatus             | [Status](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#set-status) of the Span. |
+| SpanStatus             | [Status Code](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#set-status) of the Span. |
+| SpanStatusAttributes             | [Status](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#set-status) of the Span that includes the Code and Message as a dynamic field |
```
Comment (asaharn) [inline exporter/azuredataexplorerexporter/README.md:96]:
```suggestion
| SpanStatusMessage             | [Status Message](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#set-status) of the Span |
```

**Review by ag-ramachandran — APPROVED** [review]

**Comment by ag-ramachandran** [comment]
Hello @jpkrohling , will be great if you can have a look when you get time

**Review by jpkrohling — APPROVED** [review]
Looks sane, and code owners have approved.

**Comment by ag-ramachandran** [comment]
> Looks sane, and code owners have approved.

Thanks a lot @jpkrohling. Would it be possible to merge this PR.

**Comment by jpkrohling** [comment]
CI is passing, PR merged.

**Pull request merged**